android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensions.kt

```diff
@@ -74,151 +74,176 @@ fun List<ActionConfig>.handleClickEvent(
   context: Context? = null,
 ) {
   val onClickAction =
-    this.find { it.trigger.isIn(ActionTrigger.ON_CLICK, ActionTrigger.ON_QUESTIONNAIRE_SUBMISSION) }
+    this.firstOrNull {
+      it.trigger.isIn(
+        ActionTrigger.ON_SEARCH_SINGLE_RESULT,
+        ActionTrigger.ON_CLICK,
+        ActionTrigger.ON_QUESTIONNAIRE_SUBMISSION,
+      )
+    }
 
-  onClickAction?.let { theConfig ->
-    val computedValuesMap = resourceData?.computedValuesMap ?: emptyMap()
-    val actionConfig = theConfig.interpolate(computedValuesMap)
-    val interpolatedParams = interpolateActionParamsValue(actionConfig, resourceData)
-    val practitionerId =
-      interpolatedParams
-        .find { it.paramType == ActionParameterType.RESOURCE_ID && it.key == PRACTITIONER_ID }
-        ?.value
-    val resourceId =
-      interpolatedParams.find { it.paramType == ActionParameterType.RESOURCE_ID }?.value
-        ?: resourceData?.baseResourceId
-    when (actionConfig.workflow?.let { ApplicationWorkflow.valueOf(it) }) {
-      ApplicationWorkflow.LAUNCH_QUESTIONNAIRE -> {
-        actionConfig.questionnaire?.let { questionnaireConfig ->
-          val questionnaireConfigInterpolated = questionnaireConfig.interpolate(computedValuesMap)
+  onClickAction?.handleClickEvent(navController, resourceData, navMenu, context)
+}
+
+fun ActionConfig.handleClickEvent(
+  navController: NavController,
+  resourceData: ResourceData? = null,
+  navMenu: NavigationMenuConfig? = null,
+  context: Context? = null,
+) {
+  this.takeIf {
+      it.trigger.isIn(
+        ActionTrigger.ON_SEARCH_SINGLE_RESULT,
+        ActionTrigger.ON_CLICK,
+        ActionTrigger.ON_QUESTIONNAIRE_SUBMISSION,
+      )
+    }
+    ?.let { theConfig ->
+      val computedValuesMap = resourceData?.computedValuesMap ?: emptyMap()
+      val actionConfig = theConfig.interpolate(computedValuesMap)
+      val interpolatedParams = interpolateActionParamsValue(actionConfig, resourceData)
+      val practitionerId =
+        interpolatedParams
+          .find { it.paramType == ActionParameterType.RESOURCE_ID && it.key == PRACTITIONER_ID }
+          ?.value
```

Comment:
Can you create a different action parameter type for PRACTITIONER_ID instead of reusing the RESOURCE_ID? the resource ID was meant to be used differently. Only the parameter type should be required for an action parameter. The keys are dynamic.